### PR TITLE
Convert ./components/permissions_gates/any_team_permission_gate/any_t…

### LINKED ISSUE #24776: Convert ./components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx from Class Component to Function Component
We want to move most of our codebase from Class Components to Function Components. This decision is based on [React recommendations](https://react.dev/reference/react/Component).

The task is to update the component to be a functional component, without any difference in functionality. Also, all related tests that fail may need refactoring.

Here are a few examples of the required changes:
https://github.com/mattermost/mattermost/pull/24187
https://github.com/mattermost/mattermost/pull/24210
https://github.com/mattermost/mattermost/pull/24183
https://github.com/mattermost/mattermost/pull/24181

If the tests do not work because they rely on the behavior of Class Components they also have to be migrated to functionaly test the same behavior. We prefer the use of `react-testing-library` for new tests, but it is not required.

When submitting the PR, please make sure to identify where and when the component shows, for QA to properly test the changes.

----

If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA LINK: MM-54815

### PR DESCRIPTION
**Pull request by sjcode99**
This PR convert ./components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx from Class Component to Function Component.

Fixes #24776 

```release-note
NONE
```

### CODE REVIEW

File: webapp/channels/src/components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx [hunk 1]
```diff
@@ -18,23 +17,18 @@ export type Props = {
...
      */
     children: React.ReactNode;
-}
+};
 
-export default class AnyTeamPermissionGate extends React.PureComponent<Props> {
-    static defaultProps = {
-        invert: false,
-    };
+const AnyTeamPermissionGate: React.FC<Props> = ({
```
Comment (larkox) [inline webapp/channels/src/components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx:28]: Just for coherence with the rest of the code, we prefer not having the explicit type here. Better something like:
`const Foo = ({...}: Props) => {...}`

File: webapp/channels/src/components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx [hunk 2]
```diff
@@ -18,23 +17,18 @@ export type Props = {
...
+    hasPermission,
+    invert = false,
+    children,
+}) => <>{hasPermission !== invert ? children : null}</>
```
Comment (larkox) [inline webapp/channels/src/components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx:32]: I appreciate the simplicity, but I feel having this in multiple lines is more readable. Something like this:
```suggestion
}) => (
    <>
        {hasPermission !== invert ? children : null}
    </>
);
```

File: webapp/channels/src/components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx [hunk 3]
```diff
@@ -18,23 +17,18 @@ export type Props = {
...
 
-    render() {
-        if (this.props.hasPermission !== this.props.invert) {
-            return this.props.children;
-        }
-        return null;
-    }
-}
+export default AnyTeamPermissionGate;
```
Comment (larkox) [inline webapp/channels/src/components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx:34]: This used to be a PureComponent, so add here a `React.memo`.

**Review by larkox — CHANGES_REQUESTED** [review]
Thanks for the contribution! Just a few comments.

**Comment by larkox** [comment]
@sjcode99 Also, please make sure that `npm run fix` and `npm run test` in the channels folder work fine.

**Review by larkox — APPROVED** [review]

**Comment by sjcode99** [comment]
@larkox oh!! seems it got missed!. will do it now

**Review by sinansonmez — APPROVED** [review]
Great work, thanks 🚀

**Comment by sjcode99** [comment]
Hey @larkox , I can see the PR is still not merged. Is there anything which I missed doing?

**Comment by larkox** [comment]
@sjcode99 Nothing missing from your side. Just waiting for QA Review. @lindy65 is leading that process. This may take some time (take in mind we had the weekend in the middle since I added her to this pr :smile: ) but if you don't hear from us, feel free to ping again to make sure this is not stuck.

**Comment by lindy65** [comment]
@sjcode99 @larkox - This PR hasn't been picked up by the community QAs yet so  I'll test it as soon as the test server is up

**Review by lindy65 — APPROVED** [review]
Thanks @sjcode99 - looks good to me 👍 

cc @larkox

**Comment by larkox** [comment]
Thank you for the contribution!

**Pull request merged**